### PR TITLE
Update ingress from v1beta1 to v1

### PR DESCRIPTION
**Pull request by juniorjbn**
Update ingress from v1bet1 to v1 and update `Ingress.spec.rules[0].http.paths[0].backend` changes.
https://kubernetes.io/docs/concepts/services-networking/ingress/

### CODE REVIEW

**Comment by aojea** [comment]
we discussed this some time ago, the problem is that this we are always going to break someone
ingress/v1 is 1.19 IIRC, it won't work for previous versions
ingress/v1beta1 will stop to work since 1.22 :shrug:

**Comment by BenTheElder** [comment]
> ingress/v1 is 1.19 IIRC, it won't work for previous versions
> ingress/v1beta1 will stop to work since 1.22 🤷

with 1.22 coming it might be time to switch though, we can add a note about this to the docs.
older releases have to eventually be best effort I think.

File: site/static/examples/ingress/usage.yaml [hunk 1]
```diff
@@ -46,20 +46,26 @@ spec:
...
-          serviceName: foo-service
-          servicePort: 5678
-      - path: /bar
+          service:
+            name: foo-service
+            port:
+              number: 5678
+      - pathType: Prefix
+        path: "/bar"
         backend:
```
Comment (aojea) [inline site/static/examples/ingress/usage.yaml:66]:
```suggestion
        defaultBackend:
```

Comment (aojea) [reply site/static/examples/ingress/usage.yaml:66]: ups , this is ok, sorry

File: site/static/examples/ingress/usage.yaml [hunk 2]
```diff
@@ -46,20 +46,26 @@ spec:
...
+apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
   name: example-ingress
 spec:
   rules:
   - http:
       paths:
-      - path: /foo
+      - pathType: Prefix
```
Comment (aojea) [inline site/static/examples/ingress/usage.yaml:57]: we enter here on the compatiblity problem, some ingresses need
`ImplementationSpecific`

**Comment by aojea** [comment]
> with 1.22 coming it might be time to switch though, we can add a note about this to the docs.
> older releases have to eventually be best effort I think.

this is going to be a drama XD, it took me a week to update k/k to v1, 
but this is clear the way
/ok-to-test
@juniorjbn we don't have CI , have you tried it with at least one of the ingresses of the docs?

File: site/static/examples/ingress/usage.yaml (hunk 2 above)
Comment (aojea) [reply site/static/examples/ingress/usage.yaml:57]: I can't say which is better though

**Comment by aojea** [comment]
cc @rikatz

**Comment by aojea** [comment]
at least nginx ingress is not going to work https://github.com/kubernetes/ingress-nginx/issues/7145

**Comment by rikatz** [comment]
Wanna release a beta next week :)

**Comment by BenTheElder** [comment]
Ingress Impl can't support both for some time?

**Comment by aojea** [comment]
/hold
until we have some positive feedback that the examples work

**Comment by aojea** [comment]
related https://github.com/kubernetes/ingress-nginx/issues/7305 , right now the examples doesn't work for nginx

**Comment by BenTheElder** [comment]
I think we should probably roll forward with this, people are confused about the ingress-nginx failures and kind is nominally shipping v1.21, with 1.19+ having v1 AIUI. Yes some users are still on <= v1.18, but, similarly we cannot patch bugs in Kubernetes for those versions anymore, the last patch release is done upstream since we first started talking about this issue.

File: site/static/examples/ingress/usage.yaml (hunk 2 above)
Comment (BenTheElder) [reply site/static/examples/ingress/usage.yaml:57]: we don't have to support *all* ingresses here, I'd prefer to use ingresses that can use portable upstream specs for the purposes of kind anyhow.

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

File: site/static/examples/ingress/usage.yaml (hunk 2 above)
Comment (aojea) [reply site/static/examples/ingress/usage.yaml:57]: I can't say which pathType is more portabl XD

Comment (rikatz) [reply site/static/examples/ingress/usage.yaml:57]: Prefix is more portable :)

With Fixed you have all the traffic directed to a specific path only catched :)

**Comment by BenTheElder** [comment]
bypassing github actions flakes.

**Pull request merged**